packages/react-components/react-calendar-compat/library/src/utils/dateMath/dateMath.ts

```diff
@@ -174,12 +174,18 @@ export function getDateRangeArray(
     workWeekDays = [DayOfWeek.Monday, DayOfWeek.Tuesday, DayOfWeek.Wednesday, DayOfWeek.Thursday, DayOfWeek.Friday];
   }
 
-  daysToSelectInDayView = Math.max(daysToSelectInDayView, 1);
+  daysToSelectInDayView =
+    dateRangeType === DateRangeType.Day ? daysToSelectInDayView : Math.max(daysToSelectInDayView, 1);
 
   switch (dateRangeType) {
     case DateRangeType.Day:
-      startDate = getDatePart(date);
-      endDate = addDays(startDate, daysToSelectInDayView);
+      if (daysToSelectInDayView < 0) {
+        endDate = addDays(getDatePart(date), 1);
+        startDate = addDays(endDate, daysToSelectInDayView);
+      } else {
+        startDate = getDatePart(date);
+        endDate = addDays(startDate, daysToSelectInDayView);
+      }
```

Comment:
As mentioned offline, it would be great to add tests for this. Possibly even e2e tests since we are touching v8 code.